From 1e8d7050a7d115faf78c1a7531195358c9919bf0 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Thu, 8 Jun 2017 00:23:05 +0200
Subject: [PATCH 1/1] device: check connectivity on the IP interface

curl must bind to the interface that has IP configuration, not the
underlying device. Without this commit, connectivity check fails on
certain connection types (PPPoE, WWAN).

Fixes: 9d43869e473b47542520c807dace93a6f9520964
(cherry picked from commit c66995ad4ddbe44cbbb4d4d56a48420be6d483cf)
(cherry picked from commit dc1c8c22cc8034fe4c257d43b3c31196e9899326)
---
 src/devices/nm-device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/devices/nm-device.c b/src/devices/nm-device.c
index f5eb71d..a804771 100644
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -1858,7 +1858,7 @@ nm_device_check_connectivity (NMDevice *self,
 
 		/* Kick off a real connectivity check. */
 		nm_connectivity_check_async (nm_connectivity_get (),
-		                             nm_device_get_iface (self),
+		                             nm_device_get_ip_iface (self),
 		                             concheck_cb,
 		                             data);
 		return;
-- 
2.9.4

